info.go

```diff
@@ -259,8 +282,47 @@ func (pi *ProgramInfo) Instructions() (asm.Instructions, error) {
 		return nil, fmt.Errorf("unmarshaling instructions: %w", err)
 	}
 
-	// Tag the first instruction with the name of the program, if available.
-	insns[0] = insns[0].WithSymbol(pi.Name)
+	if pi.btf != 0 {
+		btfh, err := btf.NewHandleFromID(pi.btf)
+		if err != nil {
+			return nil, fmt.Errorf("unable to get BTF handle: %w", err)
+		}
+		defer btfh.Close()
+
+		spec, err := btfh.Spec(nil)
+		if err != nil {
+			return nil, fmt.Errorf("unable to get BTF spec: %w", err)
+		}
+
+		lineInfos, err := btf.NewLineInfos(
+			bytes.NewReader(pi.lineInfos),
+			internal.NativeEndian,
+			uint32(len(pi.lineInfos))/btf.LineInfoSize,
```

Comment:
This feels a bit too "raw" to me: the user shouldn't have to care what the wire format size is. How about one of these options:

1. Take `sizedReader` instead of `io.Reader` and infer number of records in `NewLineInfos`.
2. Read records from `io.Reader` until `io.EOF` and assert that `len(lineInfos) == info2.NrLineInfos` here.

I have a slight preference for option 2 since it doesn't involve defining a custom reader interface and seems more flexible overall.

Reply:
>  assert that len(lineInfos) == info2.NrLineInfos here

`pi.lineInfos` is created as such `pi.lineInfos = make([]byte, btf.LineInfoSize*info.NrLineInfo)` so this is already true. 

> This feels a bit too "raw" to me

Yea, I get that. What we are currently doing is re-calculating `info.NrLineInfo` from the data we do have. So I suggest just adding the number of entries to `pi` so we can skip the confusing math.